File: src/interfaces/IRubiksCubeObserver.ts
import { LayerMove, Rotation } from '../engine/types'

// Implemented by whatever presents the cube (3D view, 2D view, a robot)
export default interface IRubiksCubeObserver {
  // The move that triggered the notification, so observers can present it (e.g. animate
  // the specific layer). Whole-cube re-orientations pass a Rotation; reset passes nothing.
  onMove: (move?: LayerMove | Rotation) => void
}

File: android/core/src/main/kotlin/com/mikeb/simplepuzzlecube/interfaces/IRubiksCubeObserver.kt
// Port of ../src/interfaces/IRubiksCubeObserver.ts.
package com.mikeb.simplepuzzlecube.interfaces

import com.mikeb.simplepuzzlecube.engine.Move

// The one-way contract presentations implement. The engine passes the move's identity
// so a presentation can present the specific change (animate the right layer, or the
// whole cube for a Rotation); `reset()` passes null (TS: no move argument).
fun interface IRubiksCubeObserver {
    fun onMove(move: Move?)
}
